src/wallet/wallet.cpp

```diff
@@ -4183,18 +4205,34 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(const std::string& name,
         }
         walletInstance->SetMinVersion(FEATURE_LATEST);
 
-        // generate a new seed
-        CPubKey seed = walletInstance->GenerateNewSeed();
-        if (!walletInstance->SetHDSeed(seed))
-            throw std::runtime_error(std::string(__func__) + ": Storing HD seed failed");
+        if ((wallet_creation_flags & WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
+            //selective allow to set flags
+            walletInstance->AddWalletFlag(WALLET_FLAG_DISABLE_PRIVATE_KEYS);
+        }
```

Comment:
Commit "Add per option to disable private keys during internal wallet creation"

nit `} else {`.